app/models/course_data/articles_courses.rb

```diff
@@ -67,8 +67,11 @@ def update_cache
     # We use the 'all_revisions' scope so that the dashboard system edits that
     # create sandboxes are not excluded, since those are often wind up being the
     # first edit of a mainspace article's revision history
-    self.new_article = all_revisions.exists?(new_article: true)
-
+    article_revisions = article.revisions
```

Comment:
This should probably be moved to a separate method, so that this code doesn't need to be run unless the first condition is false.

Reply:
Done!